### ISSUE TITLE
[BUG] InteractiveBrowserCredential does appear to ignore AuthenticationRecord

### ISSUE THREAD

**Issue by dragosnicolaesava** [Azure/azure-sdk-for-net#20777]
**Describe the bug**
I am expecting that passing an AuthenticationRecord, that i saved, should be enought to avoid loging in again (because this should ensure persistency across application reruns.

**Expected behavior**
When InteractiveBrowserCredential  is used it shouldn't force me to login again.

**Actual behavior (include Exception or Stack Trace)**
I always get the browser popup to login.

**To Reproduce**

Save to AuthenticationRecod to a file:
```c#
var interactiveBrowser = new InteractiveBrowserCredential(new InteractiveBrowserCredentialOptions()
{TenantId = _tenantId});
var authenticationRecord = interactiveBrowser.Authenticate();
var serializedAuthenticationRecord = JsonConvert.SerializeObject(authenticationRecord, Formatting.Indented);
File.save // (custom methods we have)...
```
Read that file and pass the AuthenticationRecord:
```c#
var stream = get stream for file (custom methods we have)..
var interactiveBrowser = new InteractiveBrowserCredential(new InteractiveBrowserCredentialOptions()
{   
    TenantId = _tenantId,
    AuthenticationRecord = AuthenticationRecord.Deserialize(stream)
});

var secretClient = new SecretClient(new Uri(url), interactiveBrowser);
```
I am using in my code the `ChainedTokenCredential` but this code should be enough to get an idea.

It is unclear if this is supposed to work (i saw that based on some examples it should work in python).

**Environment:**
package id="Azure.Identity" version="1.4.0-beta.5" targetFramework="net472"

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team members best able to assist.

**Comment by christothes** [comment]
Hi @dragosnicolaesava - Does the [sample described here ](https://github.com/Azure/azure-sdk-for-net/blob/446333a239b14ad4fe0f77de8040664da402ecde/sdk/identity/Azure.Identity/samples/ClientSideUserAuthentication.md#silent-authentication-with-authenticationrecord-and-tokencachepersistenceoptions)work for you?

**Comment by dragosnicolaesava** [comment]
Yes, the example is closed to what I was been using; 

I saw some other bugs (not certain) so I decided to try and force the users to have different ways to authenticate.
The other bug is something to do with my account being created on a tenant and working (with an invitation) on another tenant.

Maybe this is the same issue here also. Think in the case of cli I saw some case where it was mixing the tenants (same with VS), but the az cli in powershell and azure powershell work ok.
I will try to reproduce it with the full released 1.4 (I tested on the perview-5), ETA: next days.
